### PR TITLE
Handle MACRO and REPT/FOR bodies differently

### LINKED ISSUE #697: Macro and REPT/FOR body capture bugs
- [ ] Any text between `MACRO` and the newline is included in the macro body, and breaks line numbers by 1:

```
foo: MACRO println "this is handled {__LINE__}"
    println "foo {__LINE__}"
ENDM this is ignored
println "start {__LINE__}"
    foo
println "done {__LINE__}"
```

```
start $3
this is handled $1
foo $2
done $5
```

- [ ] Text from `ENDM` or `ENDR` to the newline is ignored:

```
println "start {__LINE__}"
REPT 3
    println "rept {__LINE__}"
ENDR this is ignored
println "done {__LINE__}"
```

```
start $1
repeat $3
repeat $3
repeat $3
done $5
```

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #697

### CODE REVIEW

**Comment by Rangi42** [comment]
This changes a few tests' error output, but studying the line numbers, the new output makes more sense.

**Comment by Rangi42** [comment]
Note that this fixes another error involving line numbers, where they weren't being updated by newlines in block comments.

File: src/asm/lexer.c [hunk 1]
```diff
@@ -993,10 +993,20 @@ static void discardBlockComment(void)
...
+		int c = nextChar();
+
+		switch (c) {
 		case EOF:
 			error("Unterminated block comment\n");
 			goto finish;
+		case '\r':
+		case '\n':
+			/* Handle CRLF before nextLine() since shiftChars updates colNo */
+			if (c == '\r' && peek(0) == '\n')
```
Comment (ISSOtm) [inline src/asm/lexer.c:1005]: `c == '\r'` can be implicitly done by moving it above the `case '\n':`

File: src/asm/parser.y [hunk 2]
```diff
@@ -39,6 +39,7 @@
 uint32_t nListCountEmpty;
 int32_t nPCOffset;
 bool executeElseBlock; /* If this is set, ELIFs cannot be executed anymore */
+struct CaptureBody captureBody; /* Captures a REPT/FOR or MACRO */
```
Comment (ISSOtm) [inline src/asm/parser.y:42]: Shouldn't this be `static`?

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: src/asm/parser.y (hunk 2 above)
Comment (Rangi42) [reply src/asm/parser.y:42]: I don't think so; `nListCountEmpty`, `nPCOffset`, and `executeElseBlock` aren't. They're all used only within parser.y.

Comment (ISSOtm) [reply src/asm/parser.y:42]: Well, those should be `static` as well, though that change can be made separately if you prefer.

Comment (Rangi42) [reply src/asm/parser.y]: No problem, they're `static` now.

**Review by ISSOtm — APPROVED** [review]

**Pull request merged**